File: server/model/company.js
const mongoose = require("mongoose");

const { Schema } = require("mongoose");

const CompanySchema = mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
    },

    address: {
      address: {
        type: String,
        required: true,
      },
      city: {
        type: String,
        required: true,
      },
      state: {
        type: String,
        required: true,
      },
      country: {
        type: String,
        required: true,
      },

      landMark: {
        type: String,
        required: true,
      },

      pinCode: {
        type: Number,
        required: true,
      },
    },
    phoneNumber: {
      type: Number,
      required: true,
      unique: false,
    },
    email: {
      type: String,
      required: true,
    },
    website: {
      type: String,
      required: true,
    },
    contactPersonName: {
      type: String,
      required: true,
    },
    contactPersonPhoneNumber: {
      type: Number,
      required: true,
      unique: false,
    },
    contactPersonEmail: {
      type: String,
      required: true,
    },
    departments: {
      type: Array,
      required: true,
    },
    image: {
      type: String,
    },
    tag: {
      type: Array,
    },
    displayId: {
      type: String,
    },
    deleteFlag: {
      type: Boolean,
    },
  },
  {
    timestamps: true,
  }
);
module.exports = mongoose.model("company", CompanySchema);

File: lib/models/directory_model/company_model/company.dart
/// <!-- Data Class for Company --!> ///
class Company {
  late String name;
  late Map address;
  late String phoneNumber;
  late String email;
  late String website;
  late String contactPersonName;
  late String contactPersonPhoneNumber;
  late String contactPersonEmail;
  late List departments;
  late List tag;
  String? displayId;
  String? image;
  String? sId;


  Company({
    required this.name,
    required this.address,
    required this.email,
    required this.phoneNumber,
    required this.website,
    required this.contactPersonName,
    required this.contactPersonPhoneNumber,
    required this.contactPersonEmail,
    required this.departments,
    required this.tag,
    this.image,
    this.sId,
    this.displayId = ''
  });

  Map<String, dynamic> toMap() {
    var map = <String, dynamic>{};

    map['name'] = name;
    map['address'] = address;
    map['email'] = email;
    map['phoneNumber'] = phoneNumber;
    map['website'] = website;
    map['contactPersonName'] = contactPersonName;
    map['contactPersonPhoneNumber'] = contactPersonPhoneNumber;
    map['contactPersonEmail'] = contactPersonEmail;
    map['departments'] = departments;
    map['tag'] = tag;
    map['displayId'] = displayId;
    map['image'] = image;
    map['sId'] = sId;

    return map;
  }

  Company.fromMapObject(Map<String, dynamic> map) {
    name = map['name'];
    address = map['address'];
    email = map['email'];
    website = map['website'];
    contactPersonName = map['contactPersonName'];
    contactPersonEmail = map['contactPersonEmail'];
    contactPersonPhoneNumber = map['contactPersonPhoneNumber'];
    departments = map['departments'];
    tag = map['tag'];
    displayId = map['displayId'];
    image = map['image'];
    sId = map['sId'];
  }
}
/// <!-- Data Class for Company --!> ///
